include/camp/tuple.hpp

```diff
@@ -522,6 +522,25 @@ CAMP_HOST_DEVICE constexpr auto invoke(TupleLike&& t, Fn&& f) -> decltype(
       forward<Fn>(f),
       camp::make_idx_seq_t<tuple_size<TupleLike>::value>{});
 }
+
+namespace detail
+{
+  template <class T, class Tuple, idx_t... I>
+  constexpr T make_from_tuple_impl(Tuple&& t, idx_seq<I...>)
+  {
+    return T(get<I>(forward<Tuple>(t))...);
```

Comment:
This tells me it was pasted outside the `camp` namespace, so it's resolving some things either to nothing or to the wrong thing, if you paste into the camp namespace it should work.  Alternately, we have support for setting the path to camp as an arbitrary source repo as part of RAJA's cmake setup, which might help with testing.

Reply:
Randy, how to build RAJA against an external camp is described here:

https://raja.readthedocs.io/en/master/getting_started.html

Reply:
ok...thanks. I'll give it a try directly from camp.

Reply:
@trws I am using camp directly as @rhornung67 suggested. I still need these scoped...but with the camp scope of course. I think the issue is the same as the one that I had sent to you over email.
https://stackoverflow.com/questions/35537906/call-to-function-is-ambiguous-when-irrelevant-type-defined-as-alias

I also had to scope a `move` statement. I can commit and push the changes required to compile for my use case if that is ok with you?

Reply:
@trws I can't actually push to this repo, but here are the errors:
move scoping error
xcode 11 
```
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/tuple.hpp:255:54: error: call to 'move' is ambiguous
  CAMP_HOST_DEVICE constexpr tuple(tuple&& o) : base(move(o.base)) {}
                                                     ^~~~
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/tuple.hpp:467:10: note: in instantiation of member function 'camp::tuple<const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>::tuple' requested here
  return tuple<Args&&...>(std::forward<Args>(args)...);
         ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:525:86: note: in instantiation of function template specialization 'camp::forward_as_tuple<const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>' requested here
  camp::tuple< KERNEL_CONSTRUCTOR_PARAMS &... > kernelConstructorParamsTuple = camp::forward_as_tuple( kernelConstructorParams ... );
                                                                                     ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/physicsSolvers/simplePDE/LaplaceFEM.cpp:225:5: note: in instantiation of function template specialization 'geosx::finiteElement::RegionBasedKernelApplication<RAJA::policy::loop::loop_exec, geosx::constitutive::NullModel, geosx::CellElementSubRegion, LaplaceFEMKernel, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>' requested here
    RegionBasedKernelApplication< serialPolicy,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:2310:1: note: candidate function [with _Tp = camp::internal::tuple_helper<camp::int_seq<long, 0, 1, 2, 3>, camp::list<const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &> > &]
move(_Tp&& __t) _NOEXCEPT
^
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/helpers.hpp:177:48: note: candidate function [with T = camp::internal::tuple_helper<camp::int_seq<long, 0, 1, 2, 3>, camp::list<const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &> > &]
CAMP_HOST_DEVICE constexpr type::ref::rem<T>&& move(T&& t) noexcept
```

forward_as_tuple scoping error
xcode 11
```
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/tuple.hpp:514:11: error: call to 'forward_as_tuple' is ambiguous
   return forward_as_tuple(get<Lidx>(l)..., get<Ridx>(r)...);
          ^~~~~~~~~~~~~~~~
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/tuple.hpp:524:10: note: in instantiation of function template specialization 'camp::tuple_cat_pair_forward<geosx::NodeManager &, geosx::EdgeManager &, geosx::FaceManager &, geosx::CellElementSubRegion &, const geosx::FiniteElementBase *&, geosx::constitutive::NullModel *const &, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &, 0, 1, 2, 3, 4, 5, 0, 1, 2, 3>' requested here
  return tuple_cat_pair_forward(l,              
         ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:598:59: note: in instantiation of function template specialization 'camp::tuple_cat_pair_forward<camp::tuple<geosx::NodeManager &, geosx::EdgeManager &, geosx::FaceManager &, geosx::CellElementSubRegion &, const geosx::FiniteElementBase *&, geosx::constitutive::NullModel *const &>, camp::tuple<const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &> >' requested here
          auto fullKernelComponentConstructorArgs = camp::tuple_cat_pair_forward( temp,
                                                          ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/constitutive/ConstitutivePassThru.hpp:102:7: note: in instantiation of function template specialization 'geosx::finiteElement::RegionBasedKernelApplication(geosx::MeshLevel &, const arrayView1d<const geosx::string> &, const arrayView1d<const geosx::string> &, const geosx::FiniteElementDiscretization *const, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &)::(anonymous class)::operator()(const geosx::localIndex, geosx::CellElementSubRegion &)::(anonymous class)::operator()<geosx::constitutive::NullModel>' requested here
      lambda( static_cast< NullModel * >( constitutiveRelation ) );
      ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:560:62: note: in instantiation of function template specialization 'geosx::constitutive::ConstitutivePassThru<geosx::constitutive::NullModel>::Execute<(lambda at /Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:561:71)>' requested here
    constitutive::ConstitutivePassThru< CONSTITUTIVE_BASE >::Execute( constitutiveRelation,
                                                             ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/mesh/ElementRegionManager.hpp:709:16: note: in instantiation of function template specialization 'geosx::dataRepository::Group::applyLambdaToContainer<geosx::CellElementSubRegion, geosx::ElementSubRegionBase, (lambda at /Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/mesh/ElementRegionManager.hpp:709:87)>' requested here
        Group::applyLambdaToContainer< SUBREGIONTYPE, SUBREGIONTYPES... >( subRegion, [&]( auto & castedSubRegion )
               ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/mesh/ElementRegionManager.hpp:555:5: note: in instantiation of function template specialization 'geosx::ElementRegionManager::forElementSubRegionsComplete<geosx::CellElementSubRegion, LvArray::ArrayView<const std::__1::basic_string<char>, 1, 0, long, NewChaiBuffer>, (lambda at /Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/mesh/ElementRegionManager.hpp:556:71)>' requested here
    forElementSubRegionsComplete< SUBREGIONTYPE, SUBREGIONTYPES... >( targetRegions,
    ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:529:24: note: in instantiation of function template specialization 'geosx::ElementRegionManager::forElementSubRegions<geosx::CellElementSubRegion, LvArray::ArrayView<const std::__1::basic_string<char>, 1, 0, long, NewChaiBuffer>, (lambda at /Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/finiteElement/kernelInterface/KernelBase.hpp:530:61)>' requested here
  elementRegionManager.forElementSubRegions< REGION_TYPE >( targetRegions,
                       ^
/Users/settgast1/Codes/geosx/GEOSX/src/coreComponents/physicsSolvers/simplePDE/LaplaceFEM.cpp:225:5: note: in instantiation of function template specialization 'geosx::finiteElement::RegionBasedKernelApplication<RAJA::policy::loop::loop_exec, geosx::constitutive::NullModel, geosx::CellElementSubRegion, LaplaceFEMKernel, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>' requested here
    RegionBasedKernelApplication< serialPolicy,
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/tuple:1092:1: note: candidate function [with _Tp = <geosx::NodeManager &, geosx::EdgeManager &, geosx::FaceManager &, geosx::CellElementSubRegion &, const geosx::FiniteElementBase *&, geosx::constitutive::NullModel *const &, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>]
forward_as_tuple(_Tp&&... __t) _NOEXCEPT
^
/Users/settgast1/Codes/geosx/thirdPartyLibs/install-darwin-clang-release/raja/include/camp/tuple.hpp:464:33: note: candidate function [with Args = <geosx::NodeManager &, geosx::EdgeManager &, geosx::FaceManager &, geosx::CellElementSubRegion &, const geosx::FiniteElementBase *&, geosx::constitutive::NullModel *const &, const LvArray::Array<long long, 1, camp::int_seq<long, 0>, long, NewChaiBuffer> &, geosx::EpetraMatrix &, geosx::EpetraVector &, std::__1::basic_string<char> &>]
CAMP_HOST_DEVICE constexpr auto forward_as_tuple(Args&&... args) noexcept
```

Reply:
@trws Instead of dealing with anything having to do with GEOSX, if you change the `CatPair` test s.t. the second tuple (on line 72 of `tuple.cpp`) is defined as:
```
  auto t2 = camp::make_tuple(5.1f, std::string("meh"));
```
then you will see the same error:
```
UpdateCTestConfiguration  from :/Users/settgast1/Codes/geosx/camp/build/DartConfiguration.tcl
Test project /Users/settgast1/Codes/geosx/camp/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 2
    Start 2: tuple

2: Test command: /usr/local/Cellar/cmake/3.15.2/bin/cmake "--build" "." "--target" "tuple.runner"
2: Test timeout computed to be: 10000000
2: [ 33%] Built target gtest
2: [ 66%] Built target gtest_main
2: [ 83%] Building CXX object test/CMakeFiles/tuple.dir/tuple.cpp.o
2: In file included from /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:16:
2: In file included from /Users/settgast1/Codes/geosx/camp/include/camp/camp.hpp:25:
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:487:10: error: call to 'make_tuple' is ambiguous
2:   return make_tuple(get<Lidx>(l)..., get<Ridx>(r)...);
2:          ^~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:73:13: note: in instantiation of function template specialization 'camp::tuple_cat_pair<int, char, float, std::__1::basic_string<char> , 0, 1, 0, 1>' requested here
2:   auto t3 = tuple_cat_pair(t1,
2:             ^
** HERE ARE THE CANDIDATES *******************
2:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/tuple:1084:1: note: candidate function [with _Tp = <const int &, const char &, const float &, const std::__1::basic_string<char> &>]
2: make_tuple(_Tp&&... __t)
2: ^
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:449:33: note: candidate function [with Args = <const int &, const char &, const float &, const std::__1::basic_string<char> &>]
2: CAMP_HOST_DEVICE constexpr auto make_tuple(Args&&... args)
2:                                 ^
** END HERE ARE THE CANDIDATES *****************
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:80:13: error: no matching function for call to 'tuple_cat_pair'
2:   auto t4 = tuple_cat_pair(t1, t2);
2:             ^~~~~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:491:33: note: candidate template ignored: substitution failure [with L = camp::tuple<int, char>, R = camp::tuple<float, std::__1::basic_string<char> >]: no matching function for call to 'tuple_cat_pair'
2: CAMP_HOST_DEVICE constexpr auto tuple_cat_pair(L const& l, R const& r) noexcept
2:                                 ^
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:480:33: note: candidate function template not viable: requires 4 arguments, but 2 were provided
2: CAMP_HOST_DEVICE constexpr auto tuple_cat_pair(tuple<Lelem...> const& l,
2:                                 ^
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:82:3: error: no type named 'Compare' in the global namespace
2:   ASSERT_EQ(camp::get<1>(t4), 'a');
2:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1983:63: note: expanded from macro 'GTEST_ASSERT_EQ'
2:                       EqHelper<GTEST_IS_NULL_LITERAL_(val1)>::Compare, \
2:                                                             ~~^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:23: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:                       ^~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:17: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                 ^~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:76:52: note: expanded from macro 'GTEST_ASSERT_'
2:   if (const ::testing::AssertionResult gtest_ar = (expression)) \
2:                                                    ^~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:82:3: error: expected ')'
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1982:3: note: expanded from macro 'GTEST_ASSERT_EQ'
2:   ASSERT_PRED_FORMAT2(::testing::internal:: \
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:3: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:29: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                             ^
2: <scratch space>:105:1: note: expanded from here
2: "camp::get<1>(t4)"
2: ^
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:82:3: note: to match this '('
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1982:3: note: expanded from macro 'GTEST_ASSERT_EQ'
2:   ASSERT_PRED_FORMAT2(::testing::internal:: \
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:3: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:28: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                            ^
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:83:3: error: no type named 'Compare' in the global namespace
2:   ASSERT_EQ(camp::get<2>(t4), 5.1f);
2:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1983:63: note: expanded from macro 'GTEST_ASSERT_EQ'
2:                       EqHelper<GTEST_IS_NULL_LITERAL_(val1)>::Compare, \
2:                                                             ~~^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:23: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:                       ^~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:17: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                 ^~~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:76:52: note: expanded from macro 'GTEST_ASSERT_'
2:   if (const ::testing::AssertionResult gtest_ar = (expression)) \
2:                                                    ^~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:83:3: error: expected ')'
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1982:3: note: expanded from macro 'GTEST_ASSERT_EQ'
2:   ASSERT_PRED_FORMAT2(::testing::internal:: \
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:3: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:29: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                             ^
2: <scratch space>:109:1: note: expanded from here
2: "camp::get<2>(t4)"
2: ^
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:83:3: note: to match this '('
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:2000:32: note: expanded from macro 'ASSERT_EQ'
2: # define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
2:                                ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest.h:1982:3: note: expanded from macro 'GTEST_ASSERT_EQ'
2:   ASSERT_PRED_FORMAT2(::testing::internal:: \
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:165:3: note: expanded from macro 'ASSERT_PRED_FORMAT2'
2:   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
2:   ^
2: /Users/settgast1/Codes/geosx/camp/extern/googletest/googletest/include/gtest/gtest_pred_impl.h:146:28: note: expanded from macro 'GTEST_PRED_FORMAT2_'
2:   GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), \
2:                            ^
2: In file included from /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:16:
2: In file included from /Users/settgast1/Codes/geosx/camp/include/camp/camp.hpp:25:
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:487:10: error: call to 'make_tuple' is ambiguous
2:   return make_tuple(get<Lidx>(l)..., get<Ridx>(r)...);
2:          ^~~~~~~~~~
2: /Users/settgast1/Codes/geosx/camp/test/tuple.cpp:86:7: note: in instantiation of function template specialization 'camp::tuple_cat_pair<int, char, float, std::__1::basic_string<char> , 1, 0, 1, 0>' requested here
2:       tuple_cat_pair(t1, camp::idx_seq<1, 0>{}, t2, camp::idx_seq<1, 0>{});
2:       ^
2: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/tuple:1084:1: note: candidate function [with _Tp = <const char &, const int &, const std::__1::basic_string<char> &, const float &>]
2: make_tuple(_Tp&&... __t)
2: ^
2: /Users/settgast1/Codes/geosx/camp/include/camp/tuple.hpp:449:33: note: candidate function [with Args = <const char &, const int &, const std::__1::basic_string<char> &, const float &>]
2: CAMP_HOST_DEVICE constexpr auto make_tuple(Args&&... args)
2:                                 ^
2: 7 errors generated.
2: make[3]: *** [test/CMakeFiles/tuple.dir/tuple.cpp.o] Error 1
2: make[2]: *** [test/CMakeFiles/tuple.dir/all] Error 2
2: make[1]: *** [test/CMakeFiles/tuple.runner.dir/rule] Error 2
2: make: *** [test/CMakeFiles/tuple.runner.dir/rule] Error 2
1/1 Test #2: tuple ............................***Failed    0.80 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.80 sec

The following tests FAILED:
	  2 - tuple (Failed)
Errors while running CTest
```